includes/Modules/Analytics_4.php

```diff
@@ -2326,6 +2330,27 @@ protected function inline_tag_id_mismatch( $modules_data ) {
 		return $modules_data;
 	}
 
+	/**
+	 * Populates conversion reporting events data to pass to JS via _googlesitekitModulesData.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $modules_data Inline modules data.
+	 * @return array Inline modules data.
+	 */
+	protected function inline_conversion_reporting_data( $modules_data ) {
+		if ( $this->is_connected() ) {
+			$recent_events = $this->settings->get()['recentEvents'];
+
+			// Add the data under the `analytics-4` key to make it clear it's scoped to this module.
+			// No need to check if `analytics-4` key is present, as this hook is added with higher
+			// priority than inline_custom_dimensions_data where this key is set.
+			$modules_data['analytics-4']['recentEvents'] = $recent_events;
```

Comment:
Let's check for the keys just in case; maybe the filter modified them or something else, so I think it's best to cover any cases where that key doesn't exist.